atlas/portal/pkg/svc/zserver.go

```diff
@@ -0,0 +1,219 @@
+package svc
+
+import (
+	models "atlas/portal/internal"
+	"atlas/portal/pkg/pb"
+	responderpb "atlas/responder/pkg/pb"
+	"context"
+	"errors"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"time"
+)
+
+const (
+	// version is the current version of the service
+	version            = "0.0.1"
+	invalidServiceName = "invalid service name"
+	emptyRequest       = "empty Request"
+	success            = "success"
+	errorMsg           = "error"
+	errorMissingResp   = "missing repository response error"
+	hiddenUptimeMsg    = "uptime is hidden, mode = false"
+	portal             = "portal"
+	responder          = "responder"
+	storage            = "storage"
+)
+
+type Portal struct {
+	pb.PortalServer
+	client responderpb.ResponderClient
+	logger *logrus.Logger
+	portal models.Service
+}
+
+func getPortal() models.Service {
+	return models.Service{
+		ServiceName:          "portal",
+		ServiceDesc:          "portal service desc",
+		ServiceUptime:        time.Now(),
+		ServiceCountRequests: 0,
+	}
+}
+
+func NewPortal(logger *logrus.Logger) (*Portal, error) {
+	conn, err := grpc.Dial("0.0.0.0:9091" /*+os.Getenv("PORT")*/, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		logger.Fatalf("\nDid not connect %v\n", err)
+	}
+
+	client := responderpb.NewResponderClient(conn)
+
+	return &Portal{
+		client: client,
+		logger: logger,
+		portal: getPortal(),
+	}, nil
+}
+
+func (p *Portal) GetVersion(context.Context, *empty.Empty) (*pb.VersionResponse, error) {
+	return &pb.VersionResponse{Version: version}, nil
+}
+
+func (p *Portal) GetInfo(ctx context.Context, in *pb.GetInfoRequest) (*pb.GetInfoResponse, error) {
+	if in == nil || in.Service == "" {
+		return nil, errors.New(emptyRequest)
+	}
+
+	if in.Service == portal {
+		p.portal.ServiceCountRequests++
+		return &pb.GetInfoResponse{Value: p.portal.ServiceDesc}, nil
```

Comment:
We can change the description in other places, isn't it? Should it be protected somehow here and in those other places?